### DISCUSSION TITLE
.devcontainer for pandas-profiling

### DISCUSSION THREAD

**Discussion by Anselmoo** [Data-Centric-AI-Community/fg-data-profiling#991] [Ideas]
### Missing functionality

A .devcontainer could be nice for CodeSpace Users and Organizations for quickly working on `pandas-profiling`, especially for `hotfixes` and `releases`. 

https://code.visualstudio.com/docs/remote/containers

### Proposed feature

Adding a `.devcontainer` folder including Docker-Image and `devcontainer.json`

**Comment by sbrugman** [comment]
Hi Anselm, thank you for contributing. I'm not familiar with this devcontainer.
The benefits are clear from the linked documentation, but not the need. 
What other projects are supporting this feature? Is this config needed at repo level, or can it be provided by individual developers?
Given that it's a commercial microsoft/github feature, I'm converting the issue to a discussion so that we can poll interest for this feature.

**Reply by Anselmoo** [reply]
> Hi Anselm, thank you for contributing. I'm not familiar with this devcontainer. The benefits are clear from the linked documentation, but not the need. 

> What other projects are supporting this feature? 

[AutoKERAS](https://github.com/keras-team/autokeras) is a prominent example, where I saw it for the very first time. Furthermore, they are using a specific `TensorFlow` image, and vice versa, if there is a `pandas-profiling` image, this could be used, too?! I assume.

#### A selected list of __REPOS__ using `.devcontainers` +  `mcr.microsoft.com/vscode/devcontainers/`

1. [Ruby Rails](https://github.com/rails/rails/blob/main/.devcontainer/Dockerfile)
2. [Prisma](https://github.com/prisma/prisma/blob/main/.devcontainer/Dockerfile)
3. [jhipster](https://github.com/jhipster/generator-jhipster/blob/main/generators/server/templates/devcontainer/Dockerfile.ejs)

> Is this config needed at repo level, or can it be provided by individual developers? 

The config is provided on repo level.


https://github.com/keras-team/autokeras/blob/master/.devcontainer/Dockerfile

```Dockerfile
FROM [tensorflow/tensorflow:latest](https://hub.docker.com/r/tensorflow/tensorflow)
RUN apt-get -y update
RUN apt-get -y install git vim
COPY setup.sh /setup.sh
```

They are using a MS _free_ Docker Image. In general, the MS images are coming from https://hub.docker.com/publishers/microsoftowner

Especially, in combination with `VSCode` + `Jupyter` it might have the potential to provide a ready-to-go solution for contributions without taking care of all requirements.

> Given that it's a commercial microsoft/github feature, I'm converting the issue to a discussion so that we can poll interest for this feature.

Sounds like a great idea ... 👍
